File: Archive/backend/app/Models/Tikum.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Foundation\Auth\User;
use Illuminate\Database\Eloquent\SoftDeletes;

class Tikum extends Model
{
    use HasFactory;
    use SoftDeletes;

    protected $fillable = [
        'tempat_tujuan',
        'tempat_kumpul',
        'waktu_kumpul',
        'link_group',
        'deskripsi',
        'user_id'
    ];

    public function user() {
        return $this->belongsTo(User::class);
    }
}

File: mobile/lib/model/tikum.dart
import "./user.dart";

class Tikum {
  final int userId, id;
  final String tempatTujuan, tempatKumpul;
  final String? deskripsi, linkGroup;
  final DateTime waktuKumpul;
  final User user;

  const Tikum(
      {required this.userId,
      required this.id,
      required this.tempatKumpul,
      required this.tempatTujuan,
      this.deskripsi,
      this.linkGroup,
      required this.waktuKumpul,
      this.user = const User.empty()});

  factory Tikum.fromJson(Map<String, dynamic> json) {
    var user = User.fromJson(json["user"]);
    return Tikum(
        userId: json['user_id'],
        id: json['id'],
        deskripsi: json["deskripsi"],
        linkGroup: json["link_group"],
        tempatKumpul: json["tempat_kumpul"],
        tempatTujuan: json["tempat_tujuan"],
        waktuKumpul: DateTime.parse(json["waktu_kumpul"]),
        user: user);
  }
}
